### DISCUSSION TITLE
Route Coloring and Tags Used for Mountain Biking Routes

### DISCUSSION THREAD

**Discussion by c0nsumer** [osmandapp/OsmAnd#12404] [Maps Related]
I do a great deal of mapping of mountain bike trails in OSM and OsmAnd is outstanding for mobile, offline navigation when in remote areas. When the _Show mountain bike trails_ option in _Routes_ is toggled, I can see that MTB routes are getting highlighted, but I'm having a hard time understanding the rules by which these are colored and seem to have relations highlighted.

Can someone either explain this, or point me to where in the code this is defined? I've tried searching through the code myself, but I'm having a hard time finding this.

My goals here are to both ensure that the trails I've been mapping are tagged appropriately, and perhaps work on OsmAnd to get the MTB route highlighting as useful as that in CyclOSM, mtb.waymarkedtrails.org, etc.

As an example, the routes that are part of [superroute 12425503](https://www.openstreetmap.org/relation/12425503) are colored in OsmAnd a way that doesn't quite make sense. A specific example, [way 610004303](https://www.openstreetmap.org/way/610004303#map=18/46.49664/-87.64433) has no difficulty designation ([mtb:scale:imba](https://wiki.openstreetmap.org/wiki/Key:mtb:scale:imba)) but it's part of [relation 8468010](https://www.openstreetmap.org/relation/8468010#map=16/46.4987/-87.6452) which is designated mtb:scale:imba=2 or blue square. In OsmAnd this is currently shown as a red line on the map which would _normally_ designate a Double-Black-Diamond trail.

Thanks for any pointers you can give, I'd really like to contribute to this however I can, but am lost finding where this is defined in the code. Thanks!

**Comment by vshcherb** [comment]
Hi, here is indexing part https://github.com/osmandapp/OsmAnd-resources/blob/master/obf_creation/rendering_types.xml#L1928. You can check how it is stored with OsmAndMapCreator inspector utility and here is a switch https://github.com/osmandapp/OsmAnd-resources/blob/master/rendering_styles/default.render.xml#L136 in the rendering style.

**Comment by tomeckip** [comment]
I've been updating OSM tags for the local MTB trails and noticed that they weren't showing up properly in OsmAnd. I tracked the issue down to the rendering styles OsmAnd uses. I believe the colours used by OsmAnd to display MTB IMBA difficulty don't match the colours defined by IMBA. This is related to the issue raised by the OP about the blue trails showing up as red.

OsmAnd uses the following colours: 0 - Green, 1 - Blue, 2 - Red, 3&4 - Black.
Reference: https://docs.osmand.net/docs/user/map-legend/osmand/

IMBA uses the following colours: 0 - White, 1 - Green (Circle), 2 - Blue (Square), 3 - Black (Diamond), 4 - Black (Double Diamond, but generally shown as red in other MTB maps. See Trailforks.com for example)
IMBA system reference: https://www.imba.com/sites/default/files/content/resources/2018-10/IMBATrailDifficultRatingSystem.jpg
